src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -367,30 +351,27 @@ public ReadableStreamPullState(Stream stream, CancellationToken cancellationToke
 
             _stream = stream;
             _cancellationToken = cancellationToken;
+            _buffer = new byte[65536];
         }
 
-        public async Task<Memory<byte>> ReadAsync(int desiredSize)
+        public async Task Pull()
         {
-            Memory<byte> view;
-            if (desiredSize > 0)
+            try
             {
-                if (_buffer is null || _buffer.Length < desiredSize)
+                Memory<byte> view = _buffer;
+                int length = await _stream.ReadAsync(view, _cancellationToken).ConfigureAwait(true);
+                using (Buffers.MemoryHandle handle = view.Pin())
                 {
-                    view = _buffer = new byte[desiredSize];
-                }
-                else
-                {
-                    view = _buffer.AsMemory(0, desiredSize);
+                    ReadableStreamControllerEnqueueUnsafe(this, handle, view.Length);
```

Comment:
`view.Length` should be just `length` now.

Reply:
Maybe we could remove the `Memory<byte>` now and pin the byte array manually like here:

https://github.com/dotnet/runtime/blob/76485443684db1e379fc73184d1e2549b9245db3/src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpInterop.cs#L97-L103